packages/cloudflare/src/cli/build/utils/extract-project-env-vars.ts

```diff
@@ -0,0 +1,35 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { parse } from "@dotenvx/dotenvx";
+import type { BuildOptions } from "@opennextjs/aws/build/helper.js";
+
+function readEnvFiles(fileNames: string[], { monorepoRoot, appPath }: BuildOptions) {
+  return fileNames
+    .flatMap((fileName) => [
+      ...(monorepoRoot !== appPath ? [path.join(monorepoRoot, fileName)] : []),
+      path.join(appPath, fileName),
+    ])
+    .filter((filePath) => fs.existsSync(filePath) && fs.statSync(filePath).isFile())
+    .map((filePath) => parse(fs.readFileSync(filePath).toString()));
+}
+
+/**
+ * Extracts the environment variables defined in various .env files for a project.
+ *
+ * The `NEXTJS_ENV` environment variable in `.dev.vars` determines the mode.
+ *
+ * Merged variables respect the following priority order.
```

Comment:
Maybe add a link to https://nextjs.org/docs/pages/building-your-application/configuring/environment-variables#environment-variable-load-order